pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -694,12 +713,6 @@ def _get_tools(self, model_request_parameters: ModelRequestParameters) -> list[c
     def _get_web_search_options(self, model_request_parameters: ModelRequestParameters) -> WebSearchOptions | None:
         for tool in model_request_parameters.builtin_tools:
             if isinstance(tool, WebSearchTool):  # pragma: no branch
-                if not OpenAIModelProfile.from_profile(self.profile).openai_chat_supports_web_search:
-                    raise UserError(
-                        f'WebSearchTool is not supported with `OpenAIChatModel` and model {self.model_name!r}. '
-                        f'Please use `OpenAIResponsesModel` instead.'
```

Comment:
It would be nice to keep this recommendation if we can. Maybe by doing our own in check in `prepare_request`, before the regular one?